user_tools/src/spark_rapids_pytools/rapids/rapids_tool.py

```diff
@@ -816,9 +819,39 @@ def _create_autotuner_rapids_args(self) -> list:
             return ['--auto-tuner', '--worker-info', autotuner_path]
         return []
 
+    def _get_job_submission_resources(self, tool_name: str) -> dict:
+        # Used for backward compatibility with legacy CLI that does not calculate job resources
+        submission_args = self.wrapper_options.get('jobSubmissionProps')
+        job_resources = submission_args.get('jobResources')
+        if job_resources is None:
+            # default values
+            platform_args = submission_args.get('platformArgs')
+            job_resources = Utilities.adjust_tools_resources(platform_args.get('jvmMaxHeapSize'), 1)
+        return job_resources.get(tool_name)
+
+    def _get_rapids_threads_count(self, tool_name) -> int:
+        """
+        Get the number of threads to be used by the Rapids tool
+        :return: number of threads
+        """
+        job_resources = self._get_job_submission_resources(tool_name)
+        threads_count = job_resources['rapidsThreads']
+        return ['--num-threads', f'{threads_count}']
+
+    def _re_evaluate_platform_args(self, tool_name: str) -> dict:
+        # re-evaluate the platform arguments
+        # get the job arguments
+        job_args = self.ctxt.get_ctxt('jobArgs')
+        result = copy.deepcopy(job_args)
+        job_resources = self._get_job_submission_resources(tool_name)
+        jvm_max_heap = job_resources['jvmMaxHeapSize']
+        jvm_heap_key = f'Xmx{jvm_max_heap}g'
+        result['platformArgs']['jvmArgs'].update({jvm_heap_key: ''})
```

Comment:
Could you please explain why we are doing this update?

Reply:
We are adding the jvm_heap_size here because  each concurrent RAPIDS tool can be running with different heap_size.
So, the plan was to set the vm_heap size just before the submission based on the dictionary of resources.
Note that without this update, jvmArgs won't have an entry for the `Xmx`.